prometheus.lua

```diff
@@ -521,6 +532,13 @@ end
 -- Args:
 --   sync_interval: per-worker counter sync interval (in seconds).
 function Prometheus:init_worker(sync_interval)
+  if ngx.get_phase() ~= 'init_worker' then
+    error('Prometheus:init_worker can only be called in ' ..
+      'init_worker_by_lua_block', 2)
+  end
+  if self.sync_interval and self._counter then
```

Comment:
We should throw an error here. If the counter is already initialized and user supplies different `sync_interval`, than it is not possible to modify the already configured value.

Also, we should probably warn the user when he tries to call init_worker multiple times. The code seems to handle it well, so it is not really an error, but it is not the intended use, so I think `ngx.WARN` would be appropriate.